pkg/tsdb/elasticsearch/fields_query.go

```diff
@@ -0,0 +1,190 @@
+package elasticsearch
+
+import (
+	"sort"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/plugins"
+	es "github.com/grafana/grafana/pkg/tsdb/elasticsearch/client"
+)
+
+type fieldsQuery struct {
+	client    es.Client
+	tsdbQuery plugins.DataQuery
+}
+
+var newFieldsQuery = func(client es.Client, dataQuery plugins.DataQuery) *fieldsQuery {
+	return &fieldsQuery{
+		client:    client,
+		tsdbQuery: dataQuery,
+	}
+}
+
+func (e *fieldsQuery) execute() (plugins.DataResponse, error) {
+	query := e.tsdbQuery.Queries[0]
+
+	indexMapping, err := e.client.GetIndexMapping()
+	if err != nil {
+		return plugins.DataResponse{}, err
+	}
+
+	var fieldTypeFilter string
+	if typeProp, ok := query.Model.CheckGet("fieldTypeFilter"); ok {
+		fieldTypeFilter = typeProp.MustString()
+	}
+
+	return transform(indexMapping, fieldTypeFilter, query.RefID)
+}
+
+type fieldsMap map[string]string
+
+func extractFields(fieldMap map[string]interface{}, path []string) fieldsMap {
+	// TODO: Old js implementation had this:
+	// if (configuredEsVersion < 70) {
+	// 		for (const typeName in mappings) {
+	// 	  		const properties = mappings[typeName].properties;
+	// 	  		getFieldsRecursively(properties);
+	// 		}
+	// } else {
+	// 		const properties = mappings.properties;
+	// 		getFieldsRecursively(properties);
+	// }
+	properties := fieldMap["properties"].(map[string]interface{})
+	fields := fieldsMap{}
+
+	for key, v := range properties {
+		// We skip adding the field if it's a known metadata field.
+		if isMetadataField(key) {
+			continue
+		}
+
+		vv := v.(map[string]interface{})
+
+		if vv["type"] != nil {
+			fields[strings.Join(append(path, key), ".")] = vv["type"].(string)
+		}
+
+		if vv["properties"] != nil {
+			f := extractFields(vv, append(path, key))
+
+			for k, v := range f {
+				fields[k] = v
+			}
+		}
+	}
+
+	return fields
+}
+
+// Here we transform a raw ES `/_mappings` response to a table structure to return to the frontend.
+// The table has the following structure:
+//
+// |-----------|-----------|
+// | name      | type      |
+// |-----------|-----------|
+// | fieldName | fieldType |
+// |-----------|-----------|
+func transform(indexMapping *es.IndexMappingResponse, fieldTypeFilter, refID string) (plugins.DataResponse, error) {
+	if indexMapping.Error != nil {
+		return plugins.DataResponse{
+			Results: map[string]plugins.DataQueryResult{
+				// TODO: This
+				// refID: getErrorFromElasticResponse(indexMapping.Error),
+			},
+		}, nil
+	}
+
+	// We create the table structure for the response
+	table := plugins.DataTable{
```

Comment:
What is this DataTable? didn't find reference anywhere.

Reply:
i've seen using it in https://github.com/grafana/grafana/pull/31520 and applied it here

Reply:
Hmm not what is it meant for. Like normally we should return dataFrames right? Is this only for non query data? Or is it just legacy data wrapper that should be deprecated? Because I thought that for non queries you can just return what ever json you need.

Reply:
So this query is to return autocomplete data, basically the fields from an elasticsearch index you can query, it's only used by the editor, so there's no need to use dataframes.

the reason it returns this is because we need to be compliant with the tsdb service types which expect a `plugins.DataResponse` back.